### PR TITLE
luaPackages.nfd: init at scm-1

### PR DESCRIPTION
**Pull request by spencerpogo**
###### Description of changes

https://github.com/Vexatos/nativefiledialog

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/lua-modules/nfd/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  # use zenity because default gtk impl just crashes
+  patches = [
+    (substituteAll {
+      src = ./zenity.patch;
+      inherit zenity;
+    })
+  ];
+  rockspecDir = "lua";
+
+  extraVariables.LUA_LIBDIR = "${lua5_1}/lib";
```
Comment (teto) [inline pkgs/development/lua-modules/nfd/default.nix:25]: this should use lua, not lua5_1

File: pkgs/development/lua-modules/nfd/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{ fetchFromGitHub, buildLuarocksPackage, lua, lua5_1, maintainers, pkg-config
```
Comment (teto) [inline pkgs/development/lua-modules/nfd/default.nix:1]: dont pass lua5_1, you can abort the build if lua.version is different than 5.1 though. See examples in the generated file

**Review by teto — APPROVED** [review]

**Comment by teto** [comment]
Result of `nixpkgs-review pr 185715` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by teto** [comment]
looks like errors remain
```
nix-build -A lua51Packages.nfd
error: Function called without required argument "maintainers" at /home/teto/nixpkgs3/pkgs/development/lua-modules/nfd/default.nix:1
```

**Comment by Scoder12** [comment]
```
error: attribute 'maintainers' missing

       at /home/spencer/.cache/nixpkgs-review/rev-529dd33486090a04909ccf531831f602828b052a/nixpkgs/pkgs/development/lua-modules/nfd/default.nix:39:21:

           38|     license.fullName = "zlib";
           39|     maintainers = [ lib.maintainers.scoder12 ];
             |                     ^
           40|   };

       … while querying the derivation named 'lua5.1-nfd-scm-1'
```

seems to be because of this: 
```nix
callPackage = pkgs.newScope self;
# ...
# helper functions for dealing with LUA_PATH and LUA_CPATH
lib = luaLib;  
```

**Comment by Scoder12** [comment]
```sh
❯ nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0
$ git worktree add /home/spencer/.cache/nixpkgs-review/rev-656a817620ccb8a952b9d75fcd48d105d6abd3d9/nixpkgs 08b48b1d0483096ad4af8515ce9e69fe35e3b9e5
Preparing worktree (detached HEAD 08b48b1d048)
Updating files: 100% (31319/31319), done.
HEAD is now at 08b48b1d048 Merge pull request #185846 from PedroHLC/linux_lqx-5.18.16-lqx2
$ nix-env --option system x86_64-linux -f /home/spencer/.cache/nixpkgs-review/rev-656a817620ccb8a952b9d75fcd48d105d6abd3d9/nixpkgs -qaP --xml --out-path --show-trace
$ git merge --no-commit --no-ff 656a817620ccb8a952b9d75fcd48d105d6abd3d9
Automatic merge went well; stopped before committing as requested
$ nix-env --option system x86_64-linux -f /home/spencer/.cache/nixpkgs-review/rev-656a817620ccb8a952b9d75fcd48d105d6abd3d9/nixpkgs -qaP --xml --out-path --show-trace --meta
4 packages added:
lua51Packages.nfd (init at 1) lua52Packages.nfd (init at 1) lua53Packages.nfd (init at 1) luajitPackages.nfd (init at 2.1.0-2022-04-05-nfd-scm-1)

$ nix --experimental-features nix-command build --no-link --keep-going --option build-use-sandbox relaxed -f /home/spencer/.cache/nixpkgs-review/rev-656a817620ccb8a952b9d75fcd48d105d6abd3d9/build.nix
2 packages marked as broken and skipped:
lua52Packages.nfd lua53Packages.nfd

2 packages built:
lua51Packages.nfd luajitPackages.nfd
```

File: pkgs/top-level/lua-packages.nix [hunk 3]
```diff
@@ -3,134 +3,141 @@
...
-, fetchFromGitHub, which, writeText
-, pkgs
-, lib
-}@args:
+{ fetchurl, stdenv, lua, unzip, pkg-config, pcre, oniguruma, gnulib, tre, glibc
+, sqlite, openssl, expat, autoreconfHook, gnum4, postgresql, cyrus_sasl
+, fetchFromGitHub, which, writeText, pkgs, lib }@args:
 
 let
-  packages = ( self:
```
Comment (teto) [inline pkgs/top-level/lua-packages.nix:18]: why the change in indentation ? it makes reviewing harder

Comment (Scoder12) [reply pkgs/top-level/lua-packages.nix:18]: My bad, ran `nixfmt` on it and didn't check the diff
